firebase-firestore/src/main/java/com/google/firebase/firestore/FirestoreRegistrar.java

```diff
@@ -41,13 +44,17 @@
         Component.builder(FirestoreMultiDbComponent.class)
             .add(Dependency.required(FirebaseApp.class))
             .add(Dependency.required(Context.class))
+            .add(Dependency.required(HeartBeatInfo.class))
```

Comment:
Pls switch to provider dependency, i.e. `requiredProvider()`/`optionalProvider()`, the components are not needed until later, so it's nice to not do more work than necessary(initializing the components) during Firestore initialization.